docs/RPC Encryption/index.md

```diff
@@ -0,0 +1,129 @@
+
+```xml
+<struct name="PermissionItem" since="2.0">
+    <param name="rpcName" type="String" maxlength="100" mandatory="true">
+        <description>Name of the individual RPC in the policy table.</description>
+    </param>
+    <param name="hmiPermissions" type="HMIPermissions"  mandatory="true"/>
+    <param name="parameterPermissions" type="ParameterPermissions"  mandatory="true"/>
+    <param name="requireEncryption" type="Boolean" mandatory="false" since="6.0"/>
+</struct>
+```
+Using the app level and RPC level encryption requirements, the app side will be able to determine whether the app needs encryption or not and which RPCs in particular will need encryption.
+
+An RPC message will require encryption/protection if
+- The app has `requireEncryption=true` in the `OnPermissionsChange` notification and
+- The RPC has `requireEncryption=true` in the `PermissionItem`
+
+### SDL Core
+
+SDL Core uses the `encryption_required` flags in the policy table to determine whether or not an RPC message requires encryption. 
+
+The [`encryption_required`](https://smartdevicelink.com/en/guides/sdl-overview-guides/policies/policy-fields/#encryption_required) flag, in the `functional_groupings` section of the policy table, indicates whether or not **all** the RPCs within a functional group require encryption.
+
+#### JSON Example
+
+```json
+"RemoteControl": {
+    "encryption_required" : true,
+    "rpcs": {
+        "GetInteriorVehicleData": {
+            "hmi_levels": ["BACKGROUND", "FULL", "LIMITED"]
+        },
+        "SetInteriorVehicleData": {
+            "hmi_levels": ["BACKGROUND", "FULL", "LIMITED"]
+        },
+        ...
+    }
+}
+```
+Whereas the  [`encryption_required`](https://smartdevicelink.com/en/guides/sdl-overview-guides/policies/app-policies/#encryption_required) flag, in the `app_policies` section, indicates whether the app requires encryption or not.
+
+#### JSON Example
+
+```json
+"app_policies": {
+    "default": {
+      "encryption_required": false,
+      "keep_context": false,
+      "steal_focus": false,
+      "priority": "NONE",
+      "default_hmi": "NONE",
+      "groups": [
+        "Base-4"
+      ]
+    },
+    "appid_12345": {
+      "encryption_required": true,
+      "keep_context": false,
+      "steal_focus": true,
+      "priority": "NONE",
+      "default_hmi": "NONE",
+      "groups": [
+        "Base-4", "RemoteControl"
+      ]
+    },
+    ...
+}
+```
+SDL Core informs an SDL application about the encryption requirement via an `OnPermissionsChange` notification.
+
+!!! NOTE
+Multiple functional groups can include the same RPC, each group having an `encryption_required` flag.
+
+If an app has access to multiple functional groups containing the same RPC and **at least one** of the groups requires encryption, then the RPC shall require encryption.
+!!!
+
```

Comment:
Add a note about how the encryption_required flag in the policy table translates to the RPC. Also mention the requirements about omitting from the policy table etc...

From the proposal:

```
If encryption_required=false in the app level in app_policies, sdl core and mobile proxy shall not enable rpc encryption regardless of the value of encryption_required in the function group level. SDL core does not need to check the flag in the function group level. Mobile proxy does not need to check the flag in RPC level.

If encryption_required=true or encryption_required does not exist in the app level, the flag in the function group level or RPC level shall be checked.

If encryption_required=true for a function group, all the RPCs within that function group must be sent/received in an encryption enabled SDL service. This means the app has been authenticated via TLS handshake and RPC request and response messages are encrypted. If encryption_required=false or encryption_required does not exist for a function group, the RPC messages of that function group shall not be encrypted and can be transmitted in both encryption enabled and disabled SDL services.
```